packages/cli/src/commands/onboard/auth/github/index.ts

```diff
@@ -0,0 +1,173 @@
+        github: {
+          development: {
+            clientId,
+            clientSecret,
+            ...(hasEnterprise && {
+              enterpriseInstanceUrl,
+            }),
+          },
+        },
+      },
+    },
+    catalog: {
+      locations: [
+        {
+          type: 'file',
+          target: '../../user-info.yaml',
+          rules: [{ allow: ['User'] }],
+        },
+      ],
+    },
+  };
+};
+
+type Answers = {
+  username: string;
+  clientSecret: string;
+  clientId: string;
+  hasEnterprise: boolean;
+  enterpriseInstanceUrl?: string;
+};
+
+export const github = async () => {
+  Task.log(`
+    To add GitHub authentication, you must create an OAuth App from the GitHub developer settings: ${chalk.blue(
+      'https://github.com/settings/developers',
+    )}
+    The Homepage URL should point to Backstage's frontend, while the Authorization callback URL will point to the auth backend.
+
+    Settings for local development:
+    ${chalk.cyan(`
+      Homepage URL: http://localhost:3000
+      Authorization callback URL: http://localhost:7007/api/auth/github/handler/frame`)}
+
+    You can find the full documentation page here: ${chalk.blue(
+      'https://backstage.io/docs/auth/github/provider',
+    )}
+    `);
+
+  const answers = await inquirer.prompt<Answers>([
+    {
+      type: 'input',
+      name: 'username',
+      message: 'What is your GitHub username?',
+      validate: async (input: string) => {
+        const response = await fetch(`https://api.github.com/users/${input}`);
+        if (!response.ok) {
+          return chalk.red('Unknown user. Please try again.');
+        }
+        return true;
+      },
+    },
+    {
+      type: 'input',
+      name: 'clientId',
+      message: 'What is your Client Id?',
+      validate: (input: string) => (input.length ? true : false),
+    },
+    {
+      type: 'input',
+      name: 'clientSecret',
+      message: 'What is your Client Secret?',
+      validate: (input: string) => (input.length ? true : false),
+    },
+    {
+      type: 'confirm',
+      name: 'hasEnterprise',
+      message: 'Are you using GitHub Enterprise?',
+    },
+    {
+      type: 'input',
+      name: 'enterpriseInstanceUrl',
+      message: 'What is your URL for GitHub Enterprise?',
+      when: ({ hasEnterprise }) => hasEnterprise,
+      validate: (input: string) => Boolean(new URL(input)),
+    },
+  ]);
+
+  const { username, clientId, clientSecret } = answers;
+  const config = getConfig(answers);
+
+  Task.log('Setting up GitHub Authentication for you...');
+
+  await Task.forItem(
+    'Validating',
+    'credentials',
+    async () => await validateCredentials(clientId, clientSecret),
+  );
+  await Task.forItem(
+    'Updating',
+    APP_CONFIG_FILE,
+    async () => await updateConfigFile(APP_CONFIG_FILE, config),
+  );
+  await Task.forItem(
+    'Creating',
+    USER_ENTITY_FILE,
+    async () => await addUserEntity(USER_ENTITY_FILE, username),
+  );
```

Comment:
Hmm, tbh I don't think we should encourage this. Managing users through YAML only makes sense for really small setups. Even as an onboarding step I think it's better to for example use a sign-in resolver that doesn't require users in the catalog.

Reply:
Yeah, this was initially confusion from my part not realizing the ease of just using a custom sing-in resolver. It also came up talking to @freben about our approach earlier.
We're not really sure what we want to target here, so let me keep it in just a bit longer while we figure out what we want to do. Appreciate the input 👍